lib/puppet_litmus/serverspec.rb

```diff
@@ -146,17 +152,23 @@ def run_bolt_task(task_name, params = {}, opts = {})
 
     result = run_task(task_name, target_node_name, params, config: config_data, inventory: inventory_hash)
 
-    raise "task failed\n`#{task_name}`\n======\n#{result}" if result.first['status'] != 'success' && opts[:expect_failures] != true
+    result_obj = {
+      exit_code: 0,
+      stdout: (result.first['result']['_output'] unless result.first['result']['_output'].nil?),
+      stderr: nil,
+      result: result.first['result'],
+    }
+
+    if result.first['status'] != 'success'
+      raise "task failed\n`#{task_name}`\n======\n#{result}" if opts[:expect_failures] != true
+      result_obj[:exit_code] = result.first['result']['_error']['details']['exitcode'].nil? ? 255 : result.first['result']['_error']['details']['exitcode']
```

Comment:
I don't love this ternary, could smush into another if?

Reply:
i never really liked ternery's, but that is a personal thing. this is 👍

Reply:
Would this work:

```suggestion
      result_obj[:exit_code] = result.first['result']['_error']['details'].fetch('exitcode', 255)
```

Reply:
The fetch will error if details doesn't exist (and thus, the exit code is nil). 🤔 

Might be worth looking into later, but gonna leave it as-is for now, I think?

Reply:
I think that will also be an issue with a hash:
```
[1] pry(main)> x = {}
=> {}
[2] pry(main)> x['result']['_error']['details']
NoMethodError: undefined method `[]' for nil:NilClass
from (pry):2:in `__pry__'
```

Reply:
@ekohl so digging into it a bit, that error comes whenever a prior key in the chain is missing - in your case, because there's no `result`, the error comes up from trying to index into nil for `_error`.

So! You're right, the code you submitted is exactly as guarded and shorter, so gonna flip it!

Thanks!